src/transaction/TransactionController.ts

```diff
@@ -1278,23 +1316,21 @@ export class TransactionController extends BaseController<
 
         return [meta, true];
       case TransactionStatus.submitted:
+        if (pollingAttempts === maxPollingAttempts) {
```

Comment:
Could this be moved below in the `(!txObj` block? That's where we get the third polling result. Moving it there would let us more quickly transition the status to failed. At the moment we're waiting an entire polling cycle to mark it as failed, even though we had all of the information in the last cycle.

Reply:
Thats a good point! Fixed!